client/extensions/woocommerce/app/stats/controller.js

```diff
@@ -0,0 +1,46 @@
+/**
+ * External dependencies
+ */
+import React from 'react';
+import page from 'page';
+import includes from 'lodash/includes';
+
+/**
+ * Internal dependencies
+ */
+import { renderWithReduxStore } from 'lib/react-helpers';
+import AsyncLoad from 'components/async-load';
+import StatsPagePlaceholder from 'my-sites/stats/stats-page-placeholder';
+
+function isValidParameters( context ) {
+	const validParameters = {
+		type: [ 'orders', 'customers', 'stock' ],
+		period: [ 'year', 'last_month', 'month', '7day', 'custom' ],
+		segment: [ 'sales_by_date', 'sales_by_product', 'sales_by_category', 'coupon_usage' ]
+	};
+	return Object.keys( validParameters )
+		.every( param => includes( validParameters[ param ], context.params[ param ] ) );
+}
+
+export default function StatsController( context ) {
+	if ( ! isValidParameters( context ) ) {
+		page.redirect( '/store/stats' );
```

Comment:
This doesn't append the blog id to the end. To replicate:

1. navigate to `/store/stats/orders/7day/sales_by_product/example.com`
2. change `7day` to `70dayz`
3. you get redirected to `/store/stats/`, which should be `/store/stats/example.com`